framework/web/XmlResponseFormatter.php

```diff
@@ -142,4 +143,28 @@ protected function formatScalarValue($value)
 
         return (string) $value;
     }
+
+    /**
+     * @param mixed $name
+     * @return string
+     */
+    protected function getElementName($name)
+    {
+        return (empty($name) || is_int($name) || !$this->isValidXmlName($name)) ? $this->itemTag : $name;
```

Comment:
this condition can be made more readable by assigning it to a variable with verbose naming.